src/c++/perf_analyzer/raw_data_collector.h

```diff
@@ -0,0 +1,105 @@
+// Copyright 2023, NVIDIA CORPORATION & AFFILIATES. All rights reserved.
+//
+// Redistribution and use in source and binary forms, with or without
+// modification, are permitted provided that the following conditions
+// are met:
+//  * Redistributions of source code must retain the above copyright
+//    notice, this list of conditions and the following disclaimer.
+//  * Redistributions in binary form must reproduce the above copyright
+//    notice, this list of conditions and the following disclaimer in the
+//    documentation and/or other materials provided with the distribution.
+//  * Neither the name of NVIDIA CORPORATION nor the names of its
+//    contributors may be used to endorse or promote products derived
+//    from this software without specific prior written permission.
+//
+// THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS ``AS IS'' AND ANY
+// EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+// IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR
+// PURPOSE ARE DISCLAIMED.  IN NO EVENT SHALL THE COPYRIGHT OWNER OR
+// CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL,
+// EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO,
+// PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR
+// PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY
+// OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+// (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+// OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+
+#pragma once
+
+#include <map>
+#include <tuple>
+
+#include "client_backend/client_backend.h"
+#include "constants.h"
+#include "perf_utils.h"
+#include "request_record.h"
+
+namespace triton { namespace perfanalyzer {
+
+struct PerfMode {
+  uint32_t concurrency;
+  double request_rate;
+
+  PerfMode()
+  {
+    concurrency = 0;
+    request_rate = 0.0;
+  }
+
+  PerfMode(uint64_t c, double rr)
+  {
+    concurrency = c;
+    request_rate = rr;
+  }
+
+  bool operator==(const PerfMode& rhs) const
+  {
+    return (concurrency == rhs.concurrency) &&
+           (request_rate == rhs.request_rate);
+  }
+};
+
+struct Experiment {
+  PerfMode mode;
+  std::vector<RequestRecord> requests;
+  std::vector<uint64_t> window_boundaries;
+};
+
+class RawDataCollector {
+ public:
+  static cb::Error Create(std::shared_ptr<RawDataCollector>* collector);
+  ~RawDataCollector() = default;
+
+
+  /// Add a measurement window to the collector
+  /// \param id Identifier for the experiment
+  /// \param window_start_ns The window start timestamp in nanoseconds.
+  /// \param window_end_ns The window end timestamp in nanoseconds.
+  void AddWindow(
+      PerfMode& id, uint64_t window_start_ns, uint64_t window_end_ns);
+
+  /// Add request records to an experiment
+  /// @param id Identifier for the experiment
```

Comment:
https://devblogs.microsoft.com/cppblog/visual-studio-code-c-extension-july-2020-update-doxygen-comments-and-logpoints/

It looks like `@param` is the preferred way. Since this uses it, I'll leave it. I've created a follow-up JIRA issue for converting everything else to `@...` notation.